modules/runner/src/integrationTest/java/org/apache/ignite/internal/cluster/management/ItDuplicateNodesTest.java

```diff
@@ -0,0 +1,172 @@
+import static org.apache.ignite.internal.testframework.matchers.CompletableFutureExceptionMatcher.willThrowWithCauseOrSuppressed;
+import static org.apache.ignite.internal.testframework.matchers.CompletableFutureMatcher.willCompleteSuccessfully;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.IntStream;
+import org.apache.ignite.IgniteServer;
+import org.apache.ignite.InitParameters;
+import org.apache.ignite.internal.app.IgniteServerImpl;
+import org.apache.ignite.internal.lang.IgniteStringFormatter;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.testframework.TestIgnitionManager;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.internal.testframework.WorkDirectoryExtension;
+import org.apache.ignite.network.ClusterNode;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(WorkDirectoryExtension.class)
+class ItDuplicateNodesTest extends BaseIgniteAbstractTest {
+    private static final String NODE_BOOTSTRAP_CFG_TEMPLATE = "ignite {\n"
+            + "  network: {\n"
+            + "    port: {},\n"
+            + "    nodeFinder.netClusterNodes: [ {} ]\n"
+            + "  },\n"
+            + "  clientConnector.port: {},\n"
+            + "  rest.port: {},\n"
+            + "}";
+
+    @WorkDirectory
+    private static Path WORK_DIR;
+
+    private final Map<Integer, IgniteServer> servers = new HashMap<>();
+
+    @AfterEach
+    void shutdownNodes() {
+        servers.values().forEach(IgniteServer::shutdown);
+        servers.clear();
+    }
+
+    @Test
+    void physicalTopology(TestInfo testInfo) {
+        int nodesCount = 2;
+
+        IgniteServer node1 = startEmbeddedNode(testInfo, false, 0, nodesCount);
+        IgniteServer node2 = startEmbeddedNode(testInfo, false, 1, nodesCount);
+
+        assertThat(node1.name(), is(equalTo(node2.name())));
+
+        await().untilAsserted(() -> {
+            assertThat(getPhysicalTopologyMembers(node1), hasSize(nodesCount));
+            assertThat(getPhysicalTopologyMembers(node2), hasSize(nodesCount));
+        });
+    }
+
+    @Test
+    void logicalTopology(TestInfo testInfo) {
+        int nodesCount = 3;
+
+        IgniteServer metaStorageAndCmgNode = startEmbeddedNode(testInfo, true, 0, nodesCount);
+        startEmbeddedNode(testInfo, false, 1, nodesCount);
+        startEmbeddedNode(testInfo, false, 2, nodesCount);
```

Comment:
This is pretty cryptic, it's difficult to understand which node will get which name, so who is a duplicate of whom and who is not. Is it possible to pass node name explicitly?

Also, let's not pass `TestInfo` via a parameter, even if it's needed. It can be captured by a `@BeforeEach` method and put to a field; this will reduce noise in the test methods.